lib/cequel/record/persistence.rb

```diff
@@ -82,7 +82,7 @@ def hydrate(row)
       # @since 1.0.0
       #
       def key_attributes
-        @attributes.slice(*self.class.key_column_names)
+        @cequel_attributes.slice(*self.class.key_column_names)
```

Comment:
Why the name change?

Reply:
If i recall, there was a name conflict, where the "attributes" term was exposed, and it caused problems.

Reply:
I wouldn't have changed it unless there was a reason...

Reply:
@ilanusse, what is "pretty much the case"?

It seems from <https://github.com/rails/rails/blob/5-2-stable/activemodel/lib/active_model/attributes.rb#L70>, <https://github.com/rails/rails/blob/5-2-stable/activemodel/lib/active_model/dirty.rb#L137>, etc that active model still treats `@attributes` specially. I'm worried that this change with break some of the built in activemodel code.

I think this change came from  #395. @bmurray, can you shed light on this?

Reply:
What would the test case look like that makes you comfortable with that?

Reply:
Fair question. Not sure. I suppose the existing test suite should be enough.

I'd really like to understand the original motivation for the change, though.

Reply:
At this point (it's been more than a year).. the only way to know for sure is to "undo" that change.. run the tests, and see what happens. I guess I could do that?

Reply:
I've done it. I'll write it up in a min.

Reply:
So, do I need to revert to that? Does it function, or does it have other problems? FYI.. my branch, with the other changes, as passed al the tests on Travis.. FYI.

Reply:
So... [A year ago ActiveModel inherited better attribute handling from ActiveRecord.](https://github.com/rails/rails/commit/c3675f50d2e59b7fc173d7b332860c4b1a24a726#diff-8c369514b6ba036b727449c02e58e0fb) As of that change `@attributes` needs to be an [`AttributeSet`](https://github.com/rails/rails/blob/5-2-stable/activemodel/lib/active_model/attribute_set.rb) of [`Attributes`s](https://github.com/rails/rails/blob/5-2-stable/activemodel/lib/active_model/attribute.rb) rather than a simple `Hash`. It looks like they maintained backwards compatibility with `attr_accessor` based active models by triggering the new behavior off of the existence of an `@attributes` instance variable.  

Changing our ivar name allows us to access the old behavior. It would be *way* better to adopt the new attribute approach than to simply change our instance variable name but i suppose that is a lot to ask.